### ISSUE TITLE
cabal haddock does not notice flag change

### ISSUE THREAD

**Issue by nh2** [haskell/cabal#2083]
In a package (quickcheck) we use something like:

```
{-# LANGUAGE CPP #-}

#ifdef GENERICS
-- | Random generation and shrinking of values. Can be auto derived with Generic.
#else
-- | Random generation and shrinking of values.
#endif
class Arbitrary a where ...
```

and in the cabal file

```
  if impl(ghc >= 7.2)
    cpp-options: -D_GENERICS
```

If you change the ghc version number in there, e.g. to 7.100, `cabal haddock` doesn't notice the change and passes the old flag. If you call `cabal configure` first, everything is fine.

I believe `cabal haddock` it should perform the same "did the cabal file change"-check that `cabal build` does. Like this:

```
./QuickCheck.cabal has been changed. Re-configuring with most recently used
options. If this fails, please run configure manually.
```

**Comment by mkawalec** [comment]
I'll be working on this for the next two days.

**Comment by dminuoso** [comment]
This was resolved in https://github.com/haskell/cabal/pull/2615

**Comment by 23Skidoo** [comment]
Closing as fixed by #2615
